src/BrandEmbassyCodingStandard/Sniffs/PhpUnit/ForbidSelfAssertionsSniff.php

```diff
@@ -0,0 +1,109 @@
+<?php declare(strict_types = 1);
+
+namespace BrandEmbassyCodingStandard\Sniffs\PhpUnit;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+use ReflectionMethod;
+use SlevomatCodingStandard\Helpers\NamespaceHelper;
+use SlevomatCodingStandard\Helpers\StringHelper;
+use SlevomatCodingStandard\Helpers\TokenHelper;
+use function array_map;
+use function in_array;
+use const T_EXTENDS;
+use const T_SELF;
+use const T_STRING;
+
+final class ForbidSelfAssertionsSniff implements Sniff
+{
+    private const FORBIDDEN_CLASSES = [
+        '\PHPUnit\Framework\Assert',
+    ];
+    private const FORBIDDEN_PARENTS = [
+        TestCase::class,
+    ];
+
+
+    public function register(): array
+    {
+        return [
+            T_EXTENDS,
+        ];
+    }
+
+
+    public function process(File $phpcsFile, $extendPointer): void
+    {
+        $tokens = $phpcsFile->getTokens();
+        $parentClassPointer = TokenHelper::findNext($phpcsFile, [T_STRING], $extendPointer);
+        $parentClass = NamespaceHelper::resolveClassName(
+            $phpcsFile,
+            $tokens[$parentClassPointer]['content'],
+            $extendPointer
+        );
+
+        if (!in_array($parentClass, $this->getForbiddenParents(), true)) {
+            return;
+        }
+
+        foreach (self::FORBIDDEN_CLASSES as $forbiddenClass) {
+            $this->checkForbiddenClass($phpcsFile, $extendPointer, $forbiddenClass);
+        }
+    }
+
+
+    private function checkForbiddenClass(File $phpcsFile, int $extendPointer, string $forbiddenClass): void
+    {
+        $reflection = new ReflectionClass($forbiddenClass);
+        $methods = array_map(static function (ReflectionMethod $method): string {
+            return $method->name;
+        }, $reflection->getMethods(ReflectionMethod::IS_STATIC));
+
+        $assertionTokens = $this->getAssertionPointers($phpcsFile, $extendPointer, $methods);
```

Comment:
Tady jsem se už trochu ztratil v myšlence metody a přijde mi to trochu krkolomně napsané.

Vytáhneš si všechny statické metody, pointer na `extends` a spolu s cestou k souboru to předáš metodě, která ti vrátí "assertion tokeny" a ty je třeba opravit...